vllm/spec_decode/spec_decode_worker.py

```diff
@@ -352,10 +352,15 @@ def _should_disable_all_speculation(
             self, execute_model_req: ExecuteModelRequest) -> bool:
         # When the batch size is too large, disable speculative decoding
         # to stop trading off throughput for latency.
-        disable_all_speculation = (execute_model_req.running_queue_size >=
-                                   self.disable_by_batch_size)
+        if execute_model_req.running_queue_size >= self.disable_by_batch_size:
+            return True
 
-        return disable_all_speculation
+        # Disable speculation if best_of > 1 for any sequence group
+        for seq_group_metadata in execute_model_req.seq_group_metadata_list:
+            if seq_group_metadata.sampling_params.best_of > 1:
+                return True
```

Comment:
suggest moving this to a method so you can add a docstring with more explanation on how this works (e.g. performance tradeoff)

Reply:
done